src/other-reprs.md

```diff
@@ -112,26 +112,12 @@ assert_eq!(16, size_of::<MyReprOption<&u16>>());
 
 This optimization still applies to fieldless enums with an explicit `repr(u*)`, `repr(i*)`, or `repr(C)`.
 
-## repr(packed)
+## repr(packed(n))
 
-`repr(packed)` forces Rust to strip any padding, and only align the type to a
-byte. This may improve the memory footprint, but will likely have other negative
-side-effects.
+`repr(packed(n))` forces Rust to strip any padding, and align the type to a [`n`-byte boundary][rust-reference-type-layout],
+(where `n` is a power of two). It may improve the memory footprint.
 
-In particular, most architectures *strongly* prefer values to be aligned. This
-may mean the unaligned loads are penalized (x86), or even fault (some ARM
-chips). For simple cases like directly loading or storing a packed field, the
-compiler might be able to paper over alignment issues with shifts and masks.
-However if you take a reference to a packed field, it's unlikely that the
-compiler will be able to emit code to avoid an unaligned load.
-
-[As this can cause undefined behavior][ub loads], the lint has been implemented
-and it will become a hard error.
-
-`repr(packed)` is not to be used lightly. Unless you have extreme requirements,
-this should not be used.
-
-This repr is a modifier on `repr(C)` and `repr(Rust)`.
```

Comment:
I'm not sure it is correct to remove all of this content.  Using `packed` is hazardous, and should be used with great care.  I believe the lint is still there and valid.

I believe the current state is captured in https://github.com/rust-lang/rust/issues/82523 and https://github.com/rust-lang/rust/issues/27060.  References to packed structure fields have graduated from "it's unlikely that the compiler will be able to emit code to avoid an unaligned load" to it is now a future-incompatible warning that will become an error in the future.